daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -804,6 +805,22 @@ private[lf] final class Compiler(
       }
     }
 
+  private[this] def compileContractKeyWithMaintainers(
+      tmplId: Identifier,
+      tmpl: Template,
+      tmplKey: TemplateKey,
+  ): (t.SDefinitionRef, SDefinition) = {
+    // compile a template with key into:
+    // ContractKeyWithMaintainersDefRef(tmplId) = \ <tmplArg> <token> ->
+    //   let <key> = tmplKey.body(<tmplArg>)
+    //   in { key = <key> ; maintainers = [tmplKey.maintainers] <key> }
+    topLevelFunction2(t.ContractKeyWithMaintainersDefRef(tmplId)) { (tmplArg, _, env) =>
```

Comment:
Can we gain confidence in the definition of `ContractKeyWithMaintainersDefRef` by using it in `compileToCachedContract` ?

Reply:
- `ToCachedContractDefRef` captures the lambda with the contract (or template argument) and the key instance as the lambdas 2 bound variables.

- `ContractKeyWithMaintainersDefRef` captures the lambda with the contract (or template argument) as the lambdas sole bound variable.

Whether its useful to rationalise these 2 behaviours is something I can look into as I build upon this code (I will certainly need to build cached contracts in my next steps here).

Reply:
This said, maybe some simple happy path testing would be a good idea here - if only to demo how this code is intended to be used.